tensorflow/python/ops/sparse_ops.py

```diff
@@ -2392,18 +2394,39 @@ def sparse_tensor_dense_matmul(sp_a,
       `return A*B`
   """
   # pylint: enable=line-too-long
-  sp_a = _convert_to_sparse_tensor(sp_a)
-  with ops.name_scope(name, "SparseTensorDenseMatMul",
-                      [sp_a.indices, sp_a.values, b]) as name:
-    b = ops.convert_to_tensor(b, name="b")
-    return gen_sparse_ops.sparse_tensor_dense_mat_mul(
-        a_indices=sp_a.indices,
-        a_values=sp_a.values,
-        a_shape=sp_a.dense_shape,
-        b=b,
-        adjoint_a=adjoint_a,
-        adjoint_b=adjoint_b)
 
+  if isinstance(b, sparse_tensor.SparseTensor) \
+          or isinstance(b, sparse_tensor.SparseTensorValue):
+
+    if adjoint_a == True and adjoint_b == False:
```

Comment:
`Comparison to False should be 'not expr' or 'expr is False' [singleton-comparison]`

Reply:
`Comparison to True should be just 'expr' or 'expr is True'`

Reply:
`Use implicit True/False evaluation`